AO3-7211 Add a warning to the success message when directly adding existing bookmark to collection

germpy opened the pull request:
# Pull Request Checklist

<!-- Mark steps in the checklist as complete by changing `[ ]` to `[X]` -->
* [X] Have you read ["How to write the perfect pull request"](https://github.blog/2015-01-21-how-to-write-the-perfect-pull-request/)?
* [X] Have you read the [contributing guidelines](https://github.com/otwcode/otwarchive/blob/master/CONTRIBUTING.md)?
* [X] Have you added [tests for any changed functionality](https://github.com/otwcode/otwarchive/wiki/Automated-Testing)?
* [X] Have you added the [Jira](https://otwarchive.atlassian.net) issue number
  as the *first* thing in your pull request title (e.g. `AO3-1234 Fix thing`)
* [X] Do you have fewer than 5 pull requests already open? If not, please wait
  until they are reviewed and merged before creating new pull requests.

## Issue

[https://otwarchive.atlassian.net/browse/AO3-7211](https://otwarchive.atlassian.net/browse/AO3-7211)

## Purpose

What does this PR do?
Builds on [AO3-7205](https://github.com/otwcode/otwarchive/pull/5465), but adds the warning when directly adding existing bookmark to a collection with the **Add to Collection** button. This PR adds that warning to the success message when directly adding existing bookmark to collection.

## Testing Instructions

How can the Archive's QA team verify that this is working as you intended?
Build UI locally, follow repro steps in Jira ticket AO3-7211. Video of local testing included. 

https://github.com/user-attachments/assets/a865dfe8-f8f4-4ba3-a080-f96d5a76eb2b


## References

[AO3-7205 Add a warning to the bookmark success message](https://github.com/otwcode/otwarchive/pull/5465)

## Credit
Credit under "germpy" is fine, she/her. I dunno if this is a change that really requires credit tbhhhh


[AO3-7205]: https://otwarchive.atlassian.net/browse/AO3-7205?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

germpy committed:
AO3-7211 : adds warning when adding a new to collection.

germpy committed:
Added expected warning to automated tests messages.

germpy committed:
AO3-7211: Added expected warning to automated tests messages.

germpy committed:
Merge branch 'AO3-7211' of https://github.com/germpy/otwarchive into AO3-7211

sarken commented:
Hi, germpy!

Thank you so much for this pull request. Someone will be along to review it soon.

I've updated the Jira issue status to In Review to make sure no one mistakenly creates a duplicate pull request. If you'd like the ability to comment on, assign, and transition issues in the future, you're welcome to create a Jira account! It makes things a bit easier for us on the organizational side if the Full Name on your Jira account either closely matches the name you'd like us to credit in the release notes or includes it in parentheses, e.g. "Nickname (CREDIT NAME)."

Once you've done that (or if you've already done it -- Jira has been unreliable about showing us new accounts in the admin panel lately), you can either reply here or send an email to [otw-coders@transformativeworks.org](mailto:otw-coders@transformativeworks.org) with both your account name and email address and we'll set up the permissions for you.

Thanks again for contributing! If you have any questions, you can contact us at the same email address listed above.

omerfaruk-pseud reviewed (changes requested):
Thank you so much for this PR! Two small problems, but looks pretty good.

germpy commented on app/controllers/collection_items_controller.rb:
```diff
@@ -138,6 +138,7 @@ def create
     unless new_collections.empty?
       flash[:notice] = ts("Added to collection(s): %{collections}.",
                             collections: new_collections.collect(&:title).join(", "))
+      flash[:notice] += t("bookmarks.create.warnings.private_bookmark_added_to_collection")
```
Yep! Just to clarify, should I move it to replace the empty message in line 137 (which is only assigned if new_collections and unapproved_collections are empty), or after the unless statements at around line 155?

omerfaruk-pseud replied:
The latter, line 155, (though what line 137 does is kinda the opposite of what you said.)

germpy committed:
Corrected issues:

1. Created new locale key instead of reusing bookmarks.create
2. Ensured note is added to the endings of invited and unapproved collections' success messages.

germpy committed:
Corrected issues:
1. Created new locale key instead of reusing bookmarks.create
2. Ensured note is added to the endings of invited and unapproved collections' success messages.

germpy committed:
Merge branch 'AO3-7211' of https://github.com/germpy/otwarchive into AO3-7211

omerfaruk-pseud reviewed (changes requested):
Hi! It's no problem at all!

germpy committed:
Fixed according to comments; added line break, edited 154 to use lazy lookup

omerfaruk-pseud reviewed (changes requested):
Great! Now can you add if the item to be added is a work or not to the check?
